pkg/kapp/cmd/app/factory.go

```diff
@@ -18,7 +18,7 @@ type FactorySupportObjs struct {
 	Apps                ctlapp.Apps
 }
 
-func FactoryClients(depsFactory cmdcore.DepsFactory, nsFlags cmdcore.NamespaceFlags,
+func FactoryClients(depsFactory cmdcore.DepsFactory, appNamespace string,
```

Comment:
nit: I am a bit iffy about replacing `nsFlags` with `appNamespace` here.

Primarily since we use the same factory for non-app specific commands as well. Looks like in that case we are essentially passing the value of another flag in it's stead. 

Making `app-namespace` a part of `nsFlags` might be a bit weird because of similar reasons. Maybe, we could have another argument. Maybe we should give this some thought.

Reply:
The only reason we have the nsFlags here is to get the namespace of the app (be it any command). This could be coming from the app-namespace flag or the -n flag (line 68-69)